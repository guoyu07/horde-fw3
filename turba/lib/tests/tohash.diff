001- array (
001+ Notice: Undefined offset:  1 in /usr/share/php/Horde/iCalendar.php on line 718
002-   'name' => 'Jan Schneiderö',
002+ array (
003-   'email' => 'jan@horde.org',
003+   'name' => 'Jan Schneiderö',
004-   'emails' => 'jan@horde.org',
004+   'email' => 'jan@horde.org',
005-   'nickname' => 'yunosh',
005+   'emails' => 'jan@horde.org',
006-   'alias' => 'yunosh',
006+   'nickname' => 'yunosh',
007-   'homePhone' => '+49 521 555123',
007+   'alias' => 'yunosh',
008-   'workPhone' => '+49 521 555456',
008+   'homePhone' => '+49 521 555123',
009-   'cellPhone' => '+49 177 555123',
009+   'workPhone' => '+49 521 555456',
010-   'fax' => '+49 521 555789',
010+   'cellPhone' => '+49 177 555123',
011-   'pager' => '+49 123 555789',
011+   'fax' => '+49 521 555789',
012-   'birthday' => '1971-10-01',
012+   'pager' => '+49 123 555789',
013-   'title' => 'Senior Developer (äöü)',
013+   'birthday' => '1971-10-01',
014-   'role' => 'Developer (äöü)',
014+   'title' => 'Senior Developer (äöü)',
015-   'notes' => 'A German guy (äöü)',
015+   'role' => 'Developer (äöü)',
016-   'website' => 'http://janschneider.de',
016+   'notes' => 'A German guy (äöü)',
017-   'lastname' => 'Schneiderö',
017+   'website' => 'http://janschneider.de',
018-   'firstname' => 'Jan',
018+   'lastname' => 'Schneiderö',
019-   'middlenames' => 'K.',
019+   'firstname' => 'Jan',
020-   'namePrefix' => 'Mr.',
020+   'middlenames' => 'K.',
021-   'company' => 'Horde Project',
021+   'namePrefix' => 'Mr.',
022-   'department' => 'äöü',
022+   'company' => 'Horde Project',
023-   'homeAddress' => 'Schönestr. 15
023+   'department' => 'äöü',
024- Bielefeld 33604',
024+   'homeAddress' => 'Schönestr. 15
025-   'homeStreet' => 'Schönestr. 15',
025+ Bielefeld 33604',
026-   'homeCity' => 'Bielefeld',
026+   'homeStreet' => 'Schönestr. 15',
027-   'homePostalCode' => '33604',
027+   'homeCity' => 'Bielefeld',
028-   'workAddress' => 'Hübschestr. 19
028+   'homePostalCode' => '33604',
029- Köln, Allgäu
029+   'workAddress' => 'Hübschestr. 19
030- Dänemark',
030+ Köln, Allgäu
031-   'workStreet' => 'Hübschestr. 19',
031+ Dänemark',
032-   'workCity' => 'Köln',
032+   'workStreet' => 'Hübschestr. 19',
033-   'workProvince' => 'Allgäu',
033+   'workCity' => 'Köln',
034-   'workCountry' => 'DK',
034+   'workProvince' => 'Allgäu',
035-   'timezone' => 'Europe/Berlin',
035+   'workCountry' => 'DK',
036-   'latitude' => 52.516276,
036+   'timezone' => 'Europe/Berlin',
037-   'longitude' => 13.377778,
037+   'latitude' => '0',
038- )
038+   'longitude' => '0',
039- array (
039+ )
040-   'name' => 'Jan Schneiderö',
040+ array (
041-   'email' => 'jan@horde.org',
041+   'name' => 'Jan Schneiderö',
042-   'emails' => 'jan@horde.org',
042+   'email' => 'jan@horde.org',
043-   'nickname' => 'yunosh',
043+   'emails' => 'jan@horde.org',
044-   'alias' => 'yunosh',
044+   'nickname' => 'yunosh',
045-   'homePhone' => '+49 521 555123',
045+   'alias' => 'yunosh',
046-   'workPhone' => '+49 521 555456',
046+   'homePhone' => '+49 521 555123',
047-   'cellPhone' => '+49 177 555123',
047+   'workPhone' => '+49 521 555456',
048-   'fax' => '+49 521 555789',
048+   'cellPhone' => '+49 177 555123',
049-   'pager' => '+49 123 555789',
049+   'fax' => '+49 521 555789',
050-   'birthday' => '1971-10-01',
050+   'pager' => '+49 123 555789',
051-   'title' => 'Senior Developer (äöü)',
051+   'birthday' => '1971-10-01',
052-   'role' => 'Developer (äöü)',
052+   'title' => 'Senior Developer (äöü)',
053-   'notes' => 'A German guy (äöü)',
053+   'role' => 'Developer (äöü)',
054-   'website' => 'http://janschneider.de',
054+   'notes' => 'A German guy (äöü)',
055-   'lastname' => 'Schneiderö',
055+   'website' => 'http://janschneider.de',
056-   'firstname' => 'Jan',
056+   'lastname' => 'Schneiderö',
057-   'middlenames' => 'K.',
057+   'firstname' => 'Jan',
058-   'namePrefix' => 'Mr.',
058+   'middlenames' => 'K.',
059-   'company' => 'Horde Project',
059+   'namePrefix' => 'Mr.',
060-   'department' => 'äöü',
060+   'company' => 'Horde Project',
061-   'homeAddress' => 'Schönestr. 15
061+   'department' => 'äöü',
062- Bielefeld 33604',
062+   'homeAddress' => 'Schönestr. 15
063-   'homeStreet' => 'Schönestr. 15',
063+ Bielefeld 33604',
064-   'homeCity' => 'Bielefeld',
064+   'homeStreet' => 'Schönestr. 15',
065-   'homePostalCode' => '33604',
065+   'homeCity' => 'Bielefeld',
066-   'workAddress' => 'Hübschestr. 19
066+   'homePostalCode' => '33604',
067- Köln, Allgäu
067+   'workAddress' => 'Hübschestr. 19
068- Dänemark',
068+ Köln, Allgäu
069-   'workStreet' => 'Hübschestr. 19',
069+ Dänemark',
070-   'workCity' => 'Köln',
070+   'workStreet' => 'Hübschestr. 19',
071-   'workProvince' => 'Allgäu',
071+   'workCity' => 'Köln',
072-   'workCountry' => 'DK',
072+   'workProvince' => 'Allgäu',
073-   'timezone' => 'Europe/Berlin',
073+   'workCountry' => 'DK',
074-   'latitude' => 52.516276,
074+   'timezone' => 'Europe/Berlin',
075-   'longitude' => 13.377778,
075+   'latitude' => '1',
076- )
076+   'longitude' => '1',
077- array (
077+ )
078-   'name' => 'Jan Schneider',
078+ array (
079-   'lastname' => 'Schneider',
079+   'name' => 'Jan Schneider',
080-   'firstname' => 'Jan',
080+   'lastname' => 'Schneider',
081-   'middlenames' => 'K.',
081+   'firstname' => 'Jan',
082-   'namePrefix' => 'Mr.',
082+   'middlenames' => 'K.',
083- )
083+   'namePrefix' => 'Mr.',
084- array (
084+ )
085-   'lastname' => 'Schneider',
085+ array (
086-   'firstname' => 'Jan',
086+   'lastname' => 'Schneider',
087-   'middlenames' => 'K.',
087+   'firstname' => 'Jan',
088-   'namePrefix' => 'Mr.',
088+   'middlenames' => 'K.',
089-   'name' => 'Jan Schneider',
089+   'namePrefix' => 'Mr.',
090- )
090+   'name' => 'Jan Schneider',
091- array (
091+ )
092-   'lastname' => 'Schneider',
092+ array (
093-   'firstname' => 'Jan',
093+   'lastname' => 'Schneider',
094-   'middlenames' => 'K.',
094+   'firstname' => 'Jan',
095-   'namePrefix' => 'Mr.',
095+   'middlenames' => 'K.',
096-   'name' => 'Mr. Jan K. Schneider',
096+   'namePrefix' => 'Mr.',
097- )
097+   'name' => 'Mr. Jan K. Schneider',
098- array (
098+ )
099-   'lastname' => 'B',
099+ array (
100-   'firstname' => 'A',
100+   'lastname' => 'B',
101-   'cellPhone' => '1',
101+   'firstname' => 'A',
102-   'phone' => '4',
102+   'cellPhone' => '1',
103-   'homeCellPhone' => '2',
103+   'phone' => '4',
104-   'workCellPhone' => '3',
104+   'homeCellPhone' => '2',
105-   'homePhone' => '5',
105+   'workCellPhone' => '3',
106-   'workPhone' => '6',
106+   'homePhone' => '5',
107-   'name' => 'A  B',
107+   'workPhone' => '6',
108- )
108+   'name' => 'A  B',
109- array (
109+ )
110-   'lastname' => 'Lastname',
110+ array (
111-   'firstname' => 'Firstname',
111+   'lastname' => 'Lastname',
112-   'name' => 'Lastname, Firstname',
112+   'firstname' => 'Firstname',
113-   'company' => 'Company Name',
113+   'name' => 'Lastname, Firstname',
114-   'department' => '',
114+   'company' => 'Company Name',
115-   'homePhone' => '(xxx) xxx-xxxx',
115+   'department' => '',
116-   'workPhone' => '(xxx) xxx-xxxx',
116+   'homePhone' => '(xxx) xxx-xxxx',
117-   'cellPhone' => '(xxx) xxx-xxxx',
117+   'workPhone' => '(xxx) xxx-xxxx',
118-   'email' => 'email@domain.com',
118+   'cellPhone' => '(xxx) xxx-xxxx',
119-   'emails' => 'email@domain.com',
119+   'email' => 'email@domain.com',
120-   'category' => 'Friends',
120+   'emails' => 'email@domain.com',
121-   'businessCategory' => 'Friends',
121+   'category' => 'Friends',
122-   'homeAddress' => 'Street address
122+   'businessCategory' => 'Friends',
123- City, St 12345
123+   'homeAddress' => 'Street address
124- USA',
124+ City, St 12345
125-   'homeStreet' => 'Street address',
125+ USA',
126-   'homeCity' => 'City',
126+   'homeStreet' => 'Street address',
127-   'homeProvince' => 'St',
127+   'homeCity' => 'City',
128-   'homePostalCode' => '12345',
128+   'homeProvince' => 'St',
129-   'homeCountry' => 'USA',
129+   'homePostalCode' => '12345',
130-   'workAddress' => 'Street address
130+   'homeCountry' => 'USA',
131- City, St 12345
131+   'workAddress' => 'Street address
132- USA',
132+ City, St 12345
133-   'workStreet' => 'Street address',
133+ USA',
134-   'workCity' => 'City',
134+   'workStreet' => 'Street address',
135-   'workProvince' => 'St',
135+   'workCity' => 'City',
136-   'workPostalCode' => '12345',
136+   'workProvince' => 'St',
137-   'workCountry' => 'USA',
137+   'workPostalCode' => '12345',
138- )
138+   'workCountry' => 'USA',
139- true
139+ )
140- array (
140+ true
141-   'name' => 'Jan Schneider',
141+ array (
142-   'lastname' => 'Schneider',
142+   'name' => 'Jan Schneider',
143-   'firstname' => 'Jan',
143+   'lastname' => 'Schneider',
144-   'workEmail' => 'work@example.com',
144+   'firstname' => 'Jan',
145-   'emails' => 'work@example.com, home@example.com, mail@example.com, pref@example.com',
145+   'workEmail' => 'work@example.com',
146-   'homeEmail' => 'home@example.com',
146+   'emails' => 'work@example.com, home@example.com, mail@example.com, pref@example.com',
147-   'email' => 'pref@example.com',
147+   'homeEmail' => 'home@example.com',
148- )
148+   'email' => 'pref@example.com',
149- array (
149+ )
150-   'lastname' => 'Blow',
150+ array (
151-   'firstname' => 'Joe',
151+   'lastname' => 'Blow',
152-   'name' => 'Blow, Joe',
152+   'firstname' => 'Joe',
153-   'company' => '',
153+   'name' => 'Blow, Joe',
154-   'department' => '',
154+   'company' => '',
155-   'birthday' => '1970-03-27',
155+   'department' => '',
156-   'homePhone' => '302 834 9999',
156+   'birthday' => '1970-03-27',
157-   'cellPhone' => '302 521 9999',
157+   'homePhone' => '302 834 9999',
158-   'email' => 'Blow@somwhere.net',
158+   'cellPhone' => '302 521 9999',
159-   'emails' => 'Blow@somwhere.net',
159+   'email' => 'Blow@somwhere.net',
160-   'category' => 'Personal',
160+   'emails' => 'Blow@somwhere.net',
161-   'businessCategory' => 'Personal',
161+   'category' => 'Personal',
162-   'homeAddress' => '',
162+   'businessCategory' => 'Personal',
163- )
163+   'homeAddress' => '',
164- 136
164+ )
165- array (
165+ 136
166-   'lastname' => 'McTester',
166+ array (
167-   'firstname' => 'Testie',
167+   'lastname' => 'McTester',
168-   'name' => 'Testie McTester',
168+   'firstname' => 'Testie',
169-   'company' => 'Testers Inc',
169+   'name' => 'Testie McTester',
170-   'department' => '',
170+   'company' => 'Testers Inc',
171-   'cellPhone' => '+44 775550555',
171+   'department' => '',
172-   'homePhone' => '+44 205550555',
172+   'email' => 'test@example.org',
173-   'workPhone' => '+44 205550556',
173+   'emails' => 'test@example.org, test@example.com',
174-   'homeEmail' => 'test@example.org',
174+   'commonAddress' => '111 One Street
175-   'emails' => 'test@example.org, test@example.com',
175+ London W1 1AA',
176-   'workEmail' => 'test@example.com',
176+   'commonStreet' => '111 One Street',
177-   'homeAddress' => '111 One Street
177+   'commonCity' => 'London',
178- London W1 1AA',
178+   'commonPostalCode' => 'W1 1AA',
179-   'homeStreet' => '111 One Street',
179+   'birthday' => '2008-10-08',
180-   'homeCity' => 'London',
180+   'phototype' => 'JPEG',
181-   'homePostalCode' => 'W1 1AA',
181+ )
182-   'commonAddress' => '111 One Street
183- London W1 1AA',
184-   'commonStreet' => '111 One Street',
185-   'commonCity' => 'London',
186-   'commonPostalCode' => 'W1 1AA',
187-   'birthday' => '2008-10-08',
188-   'phototype' => 'JPEG',
189- )