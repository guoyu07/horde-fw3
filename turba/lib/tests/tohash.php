<?php

require_once 'PEAR.php';
require_once 'Horde/iCalendar.php';
require dirname(__FILE__) . '/../Object.php';
require dirname(__FILE__) . '/../Driver.php';

$GLOBALS['tz']['Europe/Berlin'] = 'Europe/Berlin';
setlocale(LC_ALL, 'de_DE.ISO-8859-1');
bindtextdomain('turba', dirname(__FILE__) . '/../../locale');
textdomain('turba');

$vcard1 = '
BEGIN:VCARD
VERSION:2.1
FN;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Jan Schneider=F6
EMAIL:jan@horde.org
NICKNAME:yunosh
TEL;HOME:+49 521 555123
TEL;WORK:+49 521 555456
TEL;WORK:+49 521 999999
TEL;CELL:+49 177 555123
TEL;FAX:+49 521 555789
TEL;PAGER:+49 123 555789
BDAY:1971-10-01
TITLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Senior Developer (=E4=F6=FC)
ROLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Developer (=E4=F6=FC)
NOTE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
A German guy (=E4=F6=FC)
URL:http://janschneider.de
N;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Schneider=F6;Jan;K.;Mr.;
ORG;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Horde Project;=E4=F6=FC
ADR;HOME;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
;;Sch=F6nestr. 15;Bielefeld;;33604;
ADR;WORK;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
;;H=FCbschestr. 19;K=F6ln;Allg=E4u;;D=E4nemark
TZ;VALUE=text:+02:00; Europe/Berlin
GEO:13.377778,52.516276
BODY:
END:VCARD
';

$vcard2 = '
BEGIN:VCARD
VERSION:3.0
FN:Jan Schneiderö
EMAIL:jan@horde.org
NICKNAME:yunosh
TEL;TYPE=HOME:+49 521 555123
TEL;TYPE=WORK:+49 521 555456
TEL;TYPE=CELL:+49 177 555123
TEL;TYPE=FAX:+49 521 555789
TEL;TYPE=PAGER:+49 123 555789
BDAY:1971-10-01
TITLE:Senior Developer (äöü)
ROLE:Developer (äöü)
NOTE:A German guy (äöü)
URL:http://janschneider.de
N:Schneiderö;Jan;K.;Mr.;
ORG:Horde Project;äöü
ADR;TYPE=HOME:;;Schönestr. 15;Bielefeld;;33604;;
ADR;TYPE=WORK:;;Hübschestr. 19;Köln;Allgäu;;Dänemark
TZ;VALUE=text:+02:00; Europe/Berlin
GEO:52.516276;13.377778
BODY:
END:VCARD
';

$vcard3 = '
BEGIN:VCARD
VERSION:3.0
FN:Jan Schneider
N:Schneider;Jan;K.;Mr.;
END:VCARD
';

$vcard4 = '
BEGIN:VCARD
VERSION:3.0
N:Schneider;Jan;K.;Mr.;
END:VCARD
';

$vcard5 = '
BEGIN:VCARD
VERSION:2.1
REV:20080523T071425Z
N:B;A;;;
TEL;CELL:1
TEL;VOICE:4
X-CLASS:private
TEL;CELL;HOME:2
TEL;CELL;WORK:3
TEL;VOICE;HOME:5
TEL;VOICE;WORK:6
END:VCARD
';

$vcard6 = '
BEGIN:VCARD
VERSION:2.1
N:Lastname;Firstname;;;
FN:Lastname, Firstname
TITLE:
ORG:Company Name;
BDAY:
TEL;HOME;VOICE;X-Synthesis-Ref1:(xxx) xxx-xxxx
TEL;WORK;VOICE;X-Synthesis-Ref1:(xxx) xxx-xxxx
TEL;CELL;VOICE;X-Synthesis-Ref1:(xxx) xxx-xxxx
EMAIL:email@domain.com
URL:
CATEGORIES:Friends
NOTE;ENCODING=QUOTED-PRINTABLE:
EIN: xx-xxxxxxx
ADR;HOME:;;Street address;City;St;12345;USA
ADR;WORK:;;Street address;City;St;12345;USA
PHOTO:
END:VCARD
';

$vcard7 = '
BEGIN:VCARD
FN:Jan Schneider
N:Schneider;Jan;;;
PHOTO;ENCODING=b;TYPE=image/png:iVBORw0KGgoAAAANSUhEUgAAAAkAAAAJAgMAAACd/+6D
  AAAACVBMVEW6ABZmZmYAAACMtcxCAAAAAXRSTlMAQObYZgAAABpJREFUCFtjYACBBgYmRgEIZm
  GBYAFGMAYBABVmAOEH9qP8AAAAAElFTkSuQmCC
UID:nhCnPyv0u7
VERSION:2.1
END:VCARD
';

$vcard8 = '
BEGIN:VCARD
FN:Jan Schneider
N:Schneider;Jan;;;
EMAIL;WORK:work@example.com
EMAIL;HOME:home@example.com
EMAIL:mail@example.com
EMAIL;PREF:pref@example.com
UID:nhCnPyv0u7
VERSION:2.1
END:VCARD
';

$vcard9 = '
BEGIN:VCARD
VERSION:2.1
N:Blow;Joe;;;
FN:Blow, Joe
TITLE:
ORG:;
BDAY:19700327
TEL;HOME;VOICE;X-Synthesis-Ref1:302 834 9999
TEL;CELL;VOICE;X-Synthesis-Ref1:302 521 9999
EMAIL:Blow@somwhere.net
URL:
CATEGORIES:Personal
NOTE:
ADR;HOME:;;;;;;
PHOTO:
END:VCARD
';

// Invalid ENCODING value.
$vcard10 = '
BEGIN:VCARD
VERSION:3.0
PRODID:-//Synthesis AG//NONSGML SyncML Engine V3.1.6.10//EN
REV:20081004T151032
N:McTester;Testie;;;
FN:Testie McTester
ORG:Testers Inc;
TEL;TYPE=VOICE,CELL,X-Synthesis-Ref0:+44 775550555
TEL;TYPE=HOME,VOICE,X-Synthesis-Ref1:+44 205550555
TEL;TYPE=WORK,VOICE,X-Synthesis-Ref2:+44 205550556
EMAIL;TYPE=HOME,INTERNET,X-Synthesis-Ref0:test@example.org
EMAIL;TYPE=WORK,INTERNET,X-Synthesis-Ref1:test@example.com
ADR;TYPE=HOME,X-Synthesis-Ref0:;;111 One Street;London;;W1 1AA;
BDAY:20081008
PHOTO;TYPE=JPEG;ENCODING=BASE64:wolQTkcNChoKAAAADUlIRFIAAAAJAAAACQIDAAAAwp3
 Dv8OuwoMAAAAJUExURcK6ABZmZmYAAADCjMK1w4xCAAAAAXRSTlMAQMOmw5hmAAAAGklEQVQIW2
 NgAMKBBgYmRgEIZmHCgWABRjAGAQAVZgDDoQfDtsKjw7wAAAAASUVORMKuQmDCgg==
 
END:VCARD
';

$driver = new Turba_Driver(array());
$iCal = new Horde_iCalendar();

$iCal->parsevCalendar($vcard1);
var_export($driver->toHash($iCal->getComponent(0)));
echo "\n";

$iCal->parsevCalendar($vcard2);
var_export($driver->toHash($iCal->getComponent(0)));
echo "\n";

$iCal->parsevCalendar($vcard3);
var_export($driver->toHash($iCal->getComponent(0)));
echo "\n";

$iCal->parsevCalendar($vcard4);
var_export($driver->toHash($iCal->getComponent(0)));
echo "\n";

$driver->map['name'] = array(
    'fields' => array('namePrefix', 'firstname', 'middlenames',
                      'lastname', 'nameSuffix'),
    'attribute' => 'object_name',
    'format' => '%s %s %s %s %s');
$iCal->parsevCalendar($vcard4);
var_export($driver->toHash($iCal->getComponent(0)));
echo "\n";

$iCal->parsevCalendar($vcard5);
var_export($driver->toHash($iCal->getComponent(0)));
echo "\n";

$iCal->parsevCalendar($vcard6);
var_export($driver->toHash($iCal->getComponent(0)));
echo "\n";

$iCal->parsevCalendar($vcard7);
$hash = $driver->toHash($iCal->getComponent(0));
var_export($hash['photo'] == file_get_contents(dirname(__FILE__) . '/az.png'));
echo "\n";

$iCal->parsevCalendar($vcard8);
var_export($driver->toHash($iCal->getComponent(0)));
echo "\n";

$iCal->parsevCalendar($vcard9);
var_export($driver->toHash($iCal->getComponent(0)));
echo "\n";

$iCal->parsevCalendar($vcard10);
$hash = $driver->toHash($iCal->getComponent(0));
var_export(strlen($hash['photo']));
echo "\n";
unset($hash['photo']);
var_export($hash);

?>
