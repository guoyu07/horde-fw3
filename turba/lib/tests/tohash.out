Notice: Undefined offset:  1 in /usr/share/php/Horde/iCalendar.php on line 718
array (
  'name' => 'Jan Schneiderö',
  'email' => 'jan@horde.org',
  'emails' => 'jan@horde.org',
  'nickname' => 'yunosh',
  'alias' => 'yunosh',
  'homePhone' => '+49 521 555123',
  'workPhone' => '+49 521 555456',
  'cellPhone' => '+49 177 555123',
  'fax' => '+49 521 555789',
  'pager' => '+49 123 555789',
  'birthday' => '1971-10-01',
  'title' => 'Senior Developer (äöü)',
  'role' => 'Developer (äöü)',
  'notes' => 'A German guy (äöü)',
  'website' => 'http://janschneider.de',
  'lastname' => 'Schneiderö',
  'firstname' => 'Jan',
  'middlenames' => 'K.',
  'namePrefix' => 'Mr.',
  'company' => 'Horde Project',
  'department' => 'äöü',
  'homeAddress' => 'Schönestr. 15
Bielefeld 33604',
  'homeStreet' => 'Schönestr. 15',
  'homeCity' => 'Bielefeld',
  'homePostalCode' => '33604',
  'workAddress' => 'Hübschestr. 19
Köln, Allgäu
Dänemark',
  'workStreet' => 'Hübschestr. 19',
  'workCity' => 'Köln',
  'workProvince' => 'Allgäu',
  'workCountry' => 'DK',
  'timezone' => 'Europe/Berlin',
  'latitude' => '0',
  'longitude' => '0',
)
array (
  'name' => 'Jan Schneiderö',
  'email' => 'jan@horde.org',
  'emails' => 'jan@horde.org',
  'nickname' => 'yunosh',
  'alias' => 'yunosh',
  'homePhone' => '+49 521 555123',
  'workPhone' => '+49 521 555456',
  'cellPhone' => '+49 177 555123',
  'fax' => '+49 521 555789',
  'pager' => '+49 123 555789',
  'birthday' => '1971-10-01',
  'title' => 'Senior Developer (äöü)',
  'role' => 'Developer (äöü)',
  'notes' => 'A German guy (äöü)',
  'website' => 'http://janschneider.de',
  'lastname' => 'Schneiderö',
  'firstname' => 'Jan',
  'middlenames' => 'K.',
  'namePrefix' => 'Mr.',
  'company' => 'Horde Project',
  'department' => 'äöü',
  'homeAddress' => 'Schönestr. 15
Bielefeld 33604',
  'homeStreet' => 'Schönestr. 15',
  'homeCity' => 'Bielefeld',
  'homePostalCode' => '33604',
  'workAddress' => 'Hübschestr. 19
Köln, Allgäu
Dänemark',
  'workStreet' => 'Hübschestr. 19',
  'workCity' => 'Köln',
  'workProvince' => 'Allgäu',
  'workCountry' => 'DK',
  'timezone' => 'Europe/Berlin',
  'latitude' => '1',
  'longitude' => '1',
)
array (
  'name' => 'Jan Schneider',
  'lastname' => 'Schneider',
  'firstname' => 'Jan',
  'middlenames' => 'K.',
  'namePrefix' => 'Mr.',
)
array (
  'lastname' => 'Schneider',
  'firstname' => 'Jan',
  'middlenames' => 'K.',
  'namePrefix' => 'Mr.',
  'name' => 'Jan Schneider',
)
array (
  'lastname' => 'Schneider',
  'firstname' => 'Jan',
  'middlenames' => 'K.',
  'namePrefix' => 'Mr.',
  'name' => 'Mr. Jan K. Schneider',
)
array (
  'lastname' => 'B',
  'firstname' => 'A',
  'cellPhone' => '1',
  'phone' => '4',
  'homeCellPhone' => '2',
  'workCellPhone' => '3',
  'homePhone' => '5',
  'workPhone' => '6',
  'name' => 'A  B',
)
array (
  'lastname' => 'Lastname',
  'firstname' => 'Firstname',
  'name' => 'Lastname, Firstname',
  'company' => 'Company Name',
  'department' => '',
  'homePhone' => '(xxx) xxx-xxxx',
  'workPhone' => '(xxx) xxx-xxxx',
  'cellPhone' => '(xxx) xxx-xxxx',
  'email' => 'email@domain.com',
  'emails' => 'email@domain.com',
  'category' => 'Friends',
  'businessCategory' => 'Friends',
  'homeAddress' => 'Street address
City, St 12345
USA',
  'homeStreet' => 'Street address',
  'homeCity' => 'City',
  'homeProvince' => 'St',
  'homePostalCode' => '12345',
  'homeCountry' => 'USA',
  'workAddress' => 'Street address
City, St 12345
USA',
  'workStreet' => 'Street address',
  'workCity' => 'City',
  'workProvince' => 'St',
  'workPostalCode' => '12345',
  'workCountry' => 'USA',
)
true
array (
  'name' => 'Jan Schneider',
  'lastname' => 'Schneider',
  'firstname' => 'Jan',
  'workEmail' => 'work@example.com',
  'emails' => 'work@example.com, home@example.com, mail@example.com, pref@example.com',
  'homeEmail' => 'home@example.com',
  'email' => 'pref@example.com',
)
array (
  'lastname' => 'Blow',
  'firstname' => 'Joe',
  'name' => 'Blow, Joe',
  'company' => '',
  'department' => '',
  'birthday' => '1970-03-27',
  'homePhone' => '302 834 9999',
  'cellPhone' => '302 521 9999',
  'email' => 'Blow@somwhere.net',
  'emails' => 'Blow@somwhere.net',
  'category' => 'Personal',
  'businessCategory' => 'Personal',
  'homeAddress' => '',
)
136
array (
  'lastname' => 'McTester',
  'firstname' => 'Testie',
  'name' => 'Testie McTester',
  'company' => 'Testers Inc',
  'department' => '',
  'email' => 'test@example.org',
  'emails' => 'test@example.org, test@example.com',
  'commonAddress' => '111 One Street
London W1 1AA',
  'commonStreet' => '111 One Street',
  'commonCity' => 'London',
  'commonPostalCode' => 'W1 1AA',
  'birthday' => '2008-10-08',
  'phototype' => 'JPEG',
)